zenoh-kuksa-provider/src/main.rs

```diff
@@ -39,69 +42,59 @@ struct Args {
     config: String,
 }
 
-fn read_config(path: &str) -> Result<ProviderConfig, Box<dyn Error>> {
-    let config_str = fs::read_to_string(path)?;
-    let config = json5::from_str(&config_str)?;
+impl Args {
+    fn read_config(&self) -> Result<ProviderConfig, Box<dyn std::error::Error>> {
+        let config_str = fs::read_to_string(&self.config)?;
+        let config = json5::from_str(&config_str)?;
 
-    Ok(config)
+        Ok(config)
+    }
 }
 
-async fn handling_zenoh_subscribtion(
-    provider_config: Arc<ProviderConfig>,
+async fn handling_zenoh_subscription(
     session: Arc<Session>,
     metadata_store: MetadataStore,
     mut kuksa_client: kuksa::Client,
 ) {
-    info!("Listening on selector: {:?}", provider_config.zenoh.key_exp);
+    info!("Listening on selector: {:?}", VEHCILE_KEY_EXPR);
```

Comment:
is there a good argument why to switch to global variable here?

Reply:
well, unless the whole provider is being changed to allow devices to indicate the signals they are interested in during runtime, the key seems to be pretty static to me.